### DISCUSSION TITLE
Unable to print using Samsung M2020 monochrome laser printer

### DISCUSSION THREAD

**Discussion by devincurrie** [OpenPrinting/system-config-printer#334] [General]
Hello,

I'm trying to get the monochrome printer to work using MX Linux 23 "Libretto" KDE using the USB connection method.

The system-config.printer dialog was able to locate the printer but there's no driver available for it. Tried using the Generic driver but that doesn't work.

Below is the troubleshooting report. How do I fix the issue?

Thanks,

Devin

`Page 1 (<troubleshoot.SchedulerNotRunning.SchedulerNotRunning object at 0x7f53d38c5410>):
{'cups_connection_failure': False}
Page 2 (<troubleshoot.CheckLocalServerPublishing.CheckLocalServerPublishing object at 0x7f53d83ae790>):
{'local_server_exporting_printers': False}
Page 3 (<troubleshoot.ChoosePrinter.ChoosePrinter object at 0x7f53d3d6ae90>):
{'cups_dest': <cups.Dest Samsung-M2020-Series (default)>,
 'cups_instance': None,
 'cups_queue': 'Samsung-M2020-Series',
 'cups_queue_listed': True}
Page 4 (<troubleshoot.CheckPrinterSanity.CheckPrinterSanity object at 0x7f53d3d6a090>):
{'cups_device_uri_scheme': 'usb',
 'cups_printer_dict': {'device-uri': 'usb://Samsung/M2020%20Series?serial=070MB8GFCC00G0L',
                       'printer-info': 'Samsung M2020 Series',
                       'printer-is-shared': True,
                       'printer-location': 'mx.desktop',
                       'printer-make-and-model': 'Generic Text-Only Printer',
                       'printer-state': 3,
                       'printer-state-message': '',
                       'printer-state-reasons': ['none'],
                       'printer-type': 143364,
                       'printer-uri-supported': 'ipp://localhost/printers/Samsung-M2020-Series'},
 'cups_printer_remote': False,
 'is_cups_class': False,
 'local_cups_queue_attributes': {'charset-configured': 'utf-8',
                                 'charset-supported': ['us-ascii', 'utf-8'],
                                 'color-supported': False,
                                 'compression-supported': ['none', 'gzip'],
                                 'copies-default': 1,
                                 'copies-supported': (1, 9999),
                                 'cups-version': '2.4.2',
                                 'device-uri': 'usb://Samsung/M2020%20Series?serial=070MB8GFCC00G0L',
                                 'document-format-default': 'application/octet-stream',
                                 'document-format-preferred': 'image/urf',
                                 'document-format-supported': ['application/octet-stream',
                                                               'application/vnd.cups-raw',
                                                               'application/x-cshell',
                                                               'application/x-csource',
                                                               'application/x-perl',
                                                               'application/x-shell',
                                                               'text/css',
                                                               'text/html',
                                                               'text/plain'],
                                 'finishings-default': 3,
                                 'finishings-supported': [3],
                                 'generated-natural-language-supported': ['en-us'],
                                 'ipp-features-supported': ['subscription-object'],
                                 'ipp-versions-supported': ['1.0',
                                                            '1.1',
                                                            '2.0',
                                                            '2.1'],
                                 'ippget-event-life': 15,
                                 'job-cancel-after-default': 10800,
                                 'job-cancel-after-supported': (0, 2147483647),
                                 'job-creation-attributes-supported': ['copies',
                                                                       'finishings',
                                                                       'finishings-col',
                                                                       'ipp-attribute-fidelity',
                                                                       'job-hold-until',
                                                                       'job-name',
                                                                       'job-priority',
                                                                       'job-sheets',
                                                                       'media',
                                                                       'media-col',
                                                                       'multiple-document-handling',
                                                                       'number-up',
                                                                       'number-up-layout',
                                                                       'orientation-requested',
                                                                       'output-bin',
                                                                       'page-delivery',
                                                                       'page-ranges',
                                                                       'print-color-mode',
                                                                       'print-quality',
                                                                       'print-scaling',
                                                                       'printer-resolution',
                                                                       'sides'],
                                 'job-hold-until-default': 'no-hold',
                                 'job-hold-until-supported': ['no-hold',
                                                              'indefinite',
                                                              'day-time',
                                                              'evening',
                                                              'night',
                                                              'second-shift',
                                                              'third-shift',
                                                              'weekend'],
                                 'job-ids-supported': True,
                                 'job-k-limit': 0,
                                 'job-k-octets-supported': (0, 479595176),
                                 'job-page-limit': 0,
                                 'job-priority-default': 50,
                                 'job-priority-supported': [100],
                                 'job-quota-period': 0,
                                 'job-settable-attributes-supported': ['copies',
                                                                       'finishings',
                                                                       'job-hold-until',
                                                                       'job-name',
                                                                       'job-priority',
                                                                       'media',
                                                                       'media-col',
                                                                       'multiple-document-handling',
                                                                       'number-up',
                                                                       'output-bin',
                                                                       'orientation-requested',
                                                                       'page-ranges',
                                                                       'print-color-mode',
                                                                       'print-quality',
                                                                       'printer-resolution',
                                                                       'sides'],
                                 'job-sheets-default': ('none', 'none'),
                                 'job-sheets-supported': ['none',
                                                          'classified',
                                                          'confidential',
                                                          'form',
                                                          'secret',
                                                          'standard',
                                                          'topsecret',
                                                          'unclassified'],
                                 'jpeg-k-octets-supported': (0, 479595176),
                                 'jpeg-x-dimension-supported': (0, 65535),
                                 'jpeg-y-dimension-supported': (1, 65535),
                                 'marker-change-time': 0,
                                 'media-bottom-margin-supported': [4],
                                 'media-col-default': '(unknown IPP value tag '
                                                      '0x34)',
                                 'media-col-ready': ['(unknown IPP value tag '
                                                     '0x34)',
                                                     '(unknown IPP value tag '
                                                     '0x34)',
                                                     '(unknown IPP value tag '
                                                     '0x34)'],
                                 'media-col-supported': ['media-bottom-margin',
                                                         'media-left-margin',
                                                         'media-right-margin',
                                                         'media-size',
                                                         'media-source',
                                                         'media-top-margin',
                                                         'media-type'],
                                 'media-default': 'na_letter_8.5x11in',
                                 'media-left-margin-supported': [4],
                                 'media-ready': ['na_letter_8.5x11in',
                                                 'na_legal_8.5x14in',
                                                 'na_ledger_11x17in'],
                                 'media-right-margin-supported': [4],
                                 'media-size-supported': ['(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)',
                                                          '(unknown IPP value '
                                                          'tag 0x34)'],
                                 'media-supported': ['na_letter_8.5x11in',
                                                     'na_legal_8.5x14in',
                                                     'na_ledger_11x17in',
                                                     'custom_17x11in_17x11in',
                                                     'iso_a4_210x297mm',
                                                     'iso_a3_297x420mm',
                                                     'na_fanfold-eur_8.5x12in',
                                                     'na_foolscap_8.5x13in',
                                                     'custom_14x11in_14x11in',
                                                     'custom_14x8.5in_14x8.5in'],
                                 'media-top-margin-supported': [4],
                                 'mopria-certified': '1.3',
                                 'multiple-document-handling-supported': ['separate-documents-uncollated-copies',
                                                                          'separate-documents-collated-copies'],
                                 'multiple-document-jobs-supported': True,
                                 'multiple-operation-time-out': 900,
                                 'multiple-operation-time-out-action': 'process-job',
                                 'natural-language-configured': 'en-us',
                                 'notify-attributes-supported': ['printer-state-change-time',
                                                                 'notify-lease-expiration-time',
                                                                 'notify-subscriber-user-name'],
                                 'notify-events-default': ['job-completed'],
                                 'notify-events-supported': ['job-completed',
                                                             'job-config-changed',
                                                             'job-created',
                                                             'job-progress',
                                                             'job-state-changed',
                                                             'job-stopped',
                                                             'printer-added',
                                                             'printer-changed',
                                                             'printer-config-changed',
                                                             'printer-deleted',
                                                             'printer-finishings-changed',
                                                             'printer-media-changed',
                                                             'printer-modified',
                                                             'printer-restarted',
                                                             'printer-shutdown',
                                                             'printer-state-changed',
                                                             'printer-stopped',
                                                             'server-audit',
                                                             'server-restarted',
                                                             'server-started',
                                                             'server-stopped'],
                                 'notify-lease-duration-default': 86400,
                                 'notify-lease-duration-supported': (0,
                                                                     2147483647),
                                 'notify-max-events-supported': [100],
                                 'notify-pull-method-supported': ['ippget'],
                                 'notify-schemes-supported': ['dbus',
                                                              'mailto',
                                                              'rss'],
                                 'number-up-default': 1,
                                 'number-up-layout-supported': ['btlr',
                                                                'btrl',
                                                                'lrbt',
                                                                'lrtb',
                                                                'rlbt',
                                                                'rltb',
                                                                'tblr',
                                                                'tbrl'],
                                 'number-up-supported': [1, 2, 4, 6, 9, 16],
                                 'operations-supported': [2,
                                                          4,
                                                          5,
                                                          6,
                                                          8,
                                                          9,
                                                          10,
                                                          11,
                                                          12,
                                                          13,
                                                          16,
                                                          17,
                                                          18,
                                                          19,
                                                          20,
                                                          21,
                                                          22,
                                                          23,
                                                          24,
                                                          25,
                                                          26,
                                                          27,
                                                          28,
                                                          34,
                                                          35,
                                                          37,
                                                          38,
                                                          56,
                                                          57,
                                                          59,
                                                          16385,
                                                          16386,
                                                          16387,
                                                          16388,
                                                          16389,
                                                          16390,
                                                          16391,
                                                          16392,
                                                          16393,
                                                          16394,
                                                          16395,
                                                          16396,
                                                          16397,
                                                          16398,
                                                          16399,
                                                          16423,
                                                          14],
                                 'orientation-requested-default': None,
                                 'orientation-requested-supported': [3,
                                                                     4,
                                                                     5,
                                                                     6],
                                 'output-bin-default': 'face-down',
                                 'output-bin-supported': ['face-down'],
                                 'page-delivery-supported': ['reverse-order',
                                                             'same-order'],
                                 'page-ranges-supported': True,
                                 'pages-per-minute': 1,
                                 'pdf-k-octets-supported': (0, 479595176),
                                 'pdf-versions-supported': ['adobe-1.2',
                                                            'adobe-1.3',
                                                            'adobe-1.4',
                                                            'adobe-1.5',
                                                            'adobe-1.6',
                                                            'adobe-1.7',
                                                            'iso-19005-1_2005',
                                                            'iso-32000-1_2008',
                                                            'pwg-5102.3'],
                                 'pdl-override-supported': ['attempted'],
                                 'port-monitor': 'none',
                                 'port-monitor-supported': ['none'],
                                 'print-color-mode-default': 'monochrome',
                                 'print-color-mode-supported': ['monochrome'],
                                 'print-quality-default': 4,
                                 'print-quality-supported': [4],
                                 'print-scaling-supported': ['auto',
                                                             'auto-fit',
                                                             'fill',
                                                             'fit',
                                                             'none'],
                                 'printer-commands': 'none',
                                 'printer-config-change-date-time': '(IPP_TAG_DATE)',
                                 'printer-config-change-time': 1694631499,
                                 'printer-current-time': '(IPP_TAG_DATE)',
                                 'printer-device-id': 'MFG:Generic;MDL:Text-Only '
                                                      'Printer;DES:Generic '
                                                      'Text-Only '
                                                      'Printer;CLS:PRINTER;CMD:TXT;DRV:Dtextonly,R1,M0;',
                                 'printer-dns-sd-name': 'Samsung M2020 Series '
                                                        '@ mx',
                                 'printer-error-policy': 'retry-job',
                                 'printer-error-policy-supported': ['abort-job',
                                                                    'retry-current-job',
                                                                    'retry-job',
                                                                    'stop-printer'],
                                 'printer-geo-location': '(unknown IPP value '
                                                         'tag 0x12)',
                                 'printer-get-attributes-supported': ['document-format'],
                                 'printer-icons': 'http://localhost/icons/Samsung-M2020-Series.png',
                                 'printer-id': 2,
                                 'printer-info': 'Samsung M2020 Series',
                                 'printer-is-accepting-jobs': True,
                                 'printer-is-shared': True,
                                 'printer-is-temporary': False,
                                 'printer-location': 'mx.desktop',
                                 'printer-make-and-model': 'Generic Text-Only '
                                                           'Printer',
                                 'printer-more-info': 'http://localhost/printers/Samsung-M2020-Series',
                                 'printer-name': 'Samsung-M2020-Series',
                                 'printer-op-policy': 'default',
                                 'printer-op-policy-supported': ['authenticated',
                                                                 'default',
                                                                 'kerberos'],
                                 'printer-organization': '',
                                 'printer-organizational-unit': '',
                                 'printer-resolution-default': (300, 300, 3),
                                 'printer-resolution-supported': [(300,
                                                                   300,
                                                                   3)],
                                 'printer-settable-attributes-supported': ['printer-geo-location',
                                                                           'printer-info',
                                                                           'printer-location',
                                                                           'printer-organization',
                                                                           'printer-organizational-unit'],
                                 'printer-state': 3,
                                 'printer-state-change-date-time': '(IPP_TAG_DATE)',
                                 'printer-state-change-time': 1694631499,
                                 'printer-state-message': '',
                                 'printer-state-reasons': ['none'],
                                 'printer-strings-uri': 'http://localhost/strings/Samsung-M2020-Series.strings',
                                 'printer-type': 143364,
                                 'printer-up-time': 1694631789,
                                 'printer-uri-supported': ['ipp://localhost/printers/Samsung-M2020-Series'],
                                 'printer-uuid': 'urn:uuid:1b685b1c-9bc8-3fdb-57fa-e409ff271f75',
                                 'pwg-raster-document-resolution-supported': [(300,
                                                                               300,
                                                                               3)],
                                 'pwg-raster-document-type-supported': ['black_1',
                                                                        'sgray_8'],
                                 'queued-job-count': 0,
                                 'server-is-sharing-printers': True,
                                 'sides-default': 'one-sided',
                                 'sides-supported': ['one-sided'],
                                 'urf-supported': ['V1.4',
                                                   'CP1',
                                                   'W8',
                                                   'PQ4',
                                                   'RS300',
                                                   'FN3'],
                                 'uri-authentication-supported': ['requesting-user-name'],
                                 'uri-security-supported': ['none'],
                                 'which-jobs-supported': ['completed',
                                                          'not-completed',
                                                          'aborted',
                                                          'all',
                                                          'canceled',
                                                          'pending',
                                                          'pending-held',
                                                          'processing',
                                                          'processing-stopped']}}
Page 5 (<troubleshoot.CheckPPDSanity.CheckPPDSanity object at 0x7f53d33a95d0>):
{'cups_printer_ppd_defaults': {'General': {'OverLongLines': 'WrapAtWidth',
                                           'PageRegion': 'Letter',
                                           'PageSize': 'Letter',
                                           'Pagination': 'True',
                                           'TabWidth': '8',
                                           'page-bottom': '2',
                                           'page-left': '5',
                                           'page-right': '2',
                                           'page-top': '2'},
                               'InstallableOptions': {'11x14RotatedAvailable': 'True',
                                                      '11x14RotatedNumColumns': '132',
                                                      '11x14RotatedNumLines': '66',
                                                      'A3Available': 'True',
                                                      'A3NumColumns': '117',
                                                      'A3NumLines': '99',
                                                      'A4Available': 'True',
                                                      'A4NumColumns': '80',
                                                      'A4NumLines': '70',
                                                      'Custom1Available': 'False',
                                                      'Custom1NumColumns': '80',
                                                      'Custom1NumLines': '66',
                                                      'Custom2Available': 'False',
                                                      'Custom2NumColumns': '80',
                                                      'Custom2NumLines': '66',
                                                      'Custom3Available': 'False',
                                                      'Custom3NumColumns': '80',
                                                      'Custom3NumLines': '66',
                                                      'FanFoldGermanAvailable': 'True',
                                                      'FanFoldGermanLegalAvailable': 'True',
                                                      'FanFoldGermanLegalNumColumns': '80',
                                                      'FanFoldGermanLegalNumLines': '78',
                                                      'FanFoldGermanNumColumns': '80',
                                                      'FanFoldGermanNumLines': '72',
                                                      'LedgerAvailable': 'True',
                                                      'LedgerNumColumns': '110',
                                                      'LedgerNumLines': '102',
                                                      'LegalAvailable': 'True',
                                                      'LegalNumColumns': '80',
                                                      'LegalNumLines': '84',
                                                      'LegalRotatedAvailable': 'True',
                                                      'LegalRotatedNumColumns': '132',
                                                      'LegalRotatedNumLines': '51',
                                                      'LetterAvailable': 'True',
                                                      'LetterNumColumns': '80',
                                                      'LetterNumLines': '66',
                                                      'NewlineCharacters': 'CRLF',
                                                      'PrinterEncoding': 'ASCII',
                                                      'SendFF': 'True',
                                                      'TabloidAvailable': 'True',
                                                      'TabloidNumColumns': '160',
                                                      'TabloidNumLines': '66'}},
 'cups_printer_ppd_valid': True,
 'missing_pkgs_and_exes': ([], [])}
Page 6 (<troubleshoot.LocalOrRemote.LocalOrRemote object at 0x7f53d33a8a10>):
{'printer_is_remote': False}
Page 7 (<troubleshoot.DeviceListed.DeviceListed object at 0x7f53d33a8350>):
{'cups_device_dict': {'device-class': 'direct',
                      'device-id': 'MFG:Samsung;CMD:SPL,URF,FWV,PIC,EXT,DCU;MDL:M2020 '
                                   'Series;CLS:PRINTER;CID:SA_SPLV3_BW;MODE:SPL3,R000105;STATUS:BUSY;',
                      'device-info': 'Samsung M2020 Series',
                      'device-location': '',
                      'device-make-and-model': 'Samsung M2020 Series'}}
Page 8 (<troubleshoot.VerifyPackages.VerifyPackages object at 0x7f53d2ed1bd0>):
{'packages_verification': {'cups': ['package cups is not installed'],
                           'foomatic': ['package foomatic is not installed'],
                           'gutenprint': ['package gutenprint is not '
                                          'installed'],
                           'hpijs': ['package hpijs is not installed'],
                           'hplip': ['package hplip is not installed'],
                           'system-config-printer': ['package '
                                                     'system-config-printer is '
                                                     'not installed']}}
Page 9 (<troubleshoot.ErrorLogCheckpoint.ErrorLogCheckpoint object at 0x7f53d33ab190>):
{'cups_server_settings': {'BrowseLocalProtocols': 'dnssd',
                          'DefaultAuthType': 'Basic',
                          'ErrorPolicy': 'retry-job',
                          'IdleExitTimeout': '60',
                          'JobPrivateAccess': 'default',
                          'JobPrivateValues': 'default',
                          'MaxLogSize': '0',
                          'PageLogFormat': '',
                          'SubscriptionPrivateAccess': 'default',
                          'SubscriptionPrivateValues': 'default',
                          'WebInterface': 'Yes',
                          '_debug_logging': '0',
                          '_remote_admin': '0',
                          '_remote_any': '0',
                          '_share_printers': '1',
                          '_user_cancel_any': '0'},
 'error_log_checkpoint': 19745,
 'error_log_debug_logging_set': True}
Page 10 (<troubleshoot.PrintTestPage.PrintTestPage object at 0x7f53d34ec890>):
{'test_page_attempted': '13/Sep/2023:12:03:29 -0700',
 'test_page_completions': [(3, 'Job completed.')],
 'test_page_job_id': [3],
 'test_page_job_status': [(True,
                           3,
                           'Samsung-M2020-Series',
                           'Test Page',
                           'Completed',
                           {'attributes-charset': 'utf-8',
                            'attributes-natural-language': 'en-us',
                            'date-time-at-completed': '(IPP_TAG_DATE)',
                            'date-time-at-creation': '(IPP_TAG_DATE)',
                            'date-time-at-processing': '(IPP_TAG_DATE)',
                            'document-format': 'text/plain',
                            'document-format-supplied': 'text/plain',
                            'job-hold-until': 'no-hold',
                            'job-id': 3,
                            'job-impressions-completed': 1,
                            'job-k-octets': 1,
                            'job-media-progress': 0,
                            'job-media-sheets-completed': 1,
                            'job-more-info': 'http://localhost/jobs/3',
                            'job-name': 'Test Page',
                            'job-originating-host-name': 'localhost',
                            'job-originating-user-name': 'Devin',
                            'job-preserved': True,
                            'job-printer-state-message': '',
                            'job-printer-state-reasons': ['none'],
                            'job-printer-up-time': 1694631836,
                            'job-printer-uri': 'ipp://localhost/printers/Samsung-M2020-Series',
                            'job-priority': 50,
                            'job-sheets': ['none', 'none'],
                            'job-state': 9,
                            'job-state-reasons': 'processing-to-stop-point',
                            'job-uri': 'ipp://localhost/jobs/3',
                            'job-uuid': 'urn:uuid:e2792f44-b78e-37f4-5e79-02ad39995ede',
                            'number-of-documents': 1,
                            'printer-uri': 'ipp://localhost/printers/Samsung-M2020-Series',
                            'time-at-completed': 1694631818,
                            'time-at-creation': 1694631809,
                            'time-at-processing': 1694631809})],
 'test_page_successful': False}
Page 11 (<troubleshoot.ErrorLogFetch.ErrorLogFetch object at 0x7f53d8b67a90>):
{'error_log_debug_logging_unset': True}
Page 12 (<troubleshoot.Locale.Locale object at 0x7f53d3ea5810>):
{'printer_page_size': 'Letter',
 'system_locale_lang': None,
 'user_locale_ctype': 'en_US',
 'user_locale_messages': 'en_US'}
`

**Comment by zdohnal** [comment]
Duplicate, closing.

**Comment by PierceDong** [comment]
https://support.hp.com/cn-zh/drivers/samsung-xpress-sl-m2020-laser-printer-series/16462592

There is uld driver, may it will work, please try.

Samsung Print Driver for Linux
